Planner: Wrong name of api permissions for PlannerBucket and PlannerPlan

Api Permissions for PlannerPlan and PlannerBucket reads `"Tasks.Read"` and `"Tasks.ReadWrite"`, but should be `"Tasks.Read.All"` and `"Tasks.ReadWrite.All"` - they're missing `".All"` at the end:  
```
{
    "resourceName": "PlannerBucket",
    "description": "",
    "permissions": {
        "graph": {
            "delegated": {
                "read": [
                    {
                        "name": "Tasks.Read"
                    }
                ],
                "update": [
                    {
                        "name": "Tasks.ReadWrite"
                    }
                ]
            },
            ...
```


#### Details of the scenario you tried and the problem that is occurring
Trying to update service principal: 
```powershell
Update-M365DSCAzureAdApplication -ApplicationName "$Name" -Permissions @(
        @{Api='Graph';PermissionName='Tasks.ReadWrite'},
        @{Api='Graph';PermissionName='Tasks.Read'}) -AdminConsent 
```
#### Verbose logs showing the problem
The following is output to the screen: 
```
2023-01-24 13:14:20 - Checking specified parameters
2023-01-24 13:14:20 -   Using a Secret as credential
2023-01-24 13:14:47 -
2023-01-24 13:14:47 - Checking existance of AD Application
2023-01-24 13:14:47 -   Application '<appname>' already exists!
2023-01-24 13:14:47 -
2023-01-24 13:14:47 - Checking app permissions
2023-01-24 13:14:47 -   Checking permission 'Graph\Tasks.ReadWrite'
2023-01-24 13:14:47 -   [ERROR] Permission 'Tasks.ReadWrite' not found!
2023-01-24 13:14:47 -   Checking permission 'Graph\Tasks.Read'
2023-01-24 13:14:47 -   [ERROR] Permission 'Tasks.Read' not found!
2023-01-24 13:14:47 -
2023-01-24 13:14:47 - Waiting 10 seconds for application creation
2023-01-24 13:14:47 -   ...
```

#### Suggested solution to the issue
Change 
- "Tasks.Read" to "Tasks.Read.All" 
- "Tasks.ReadWrite" to "Tasks.ReadWrite.All"

#### The DSC configuration that is used to reproduce the issue (as detailed as possible)
Just running the following: 
```powershell
Update-M365DSCAzureAdApplication -ApplicationName "$Name" -Permissions @(
        @{Api='Graph';PermissionName='Tasks.ReadWrite'},
        @{Api='Graph';PermissionName='Tasks.Read'}) -AdminConsent 
```

#### The operating system the target node is running
n/a

#### Version of the DSC module that was used ('dev' if using current dev branch)
Dev

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -33,6 +33,9 @@
   * Fixes #2819
 * DEPENDENCIES
   * Updated Microsoft.PowerApps.Administration.PowerShell to version 2.0.155.
+* Planner
+  * Fixed api-permissions on PlannerBucket and PlannerPlan.
+    FIXES [#2843](https://github.com/microsoft/Microsoft365DSC/issues/2843)
 
 # 1.23.118.1
 

--- a/Modules/Microsoft365DSC/DSCResources/MSFT_PlannerBucket/settings.json
+++ b/Modules/Microsoft365DSC/DSCResources/MSFT_PlannerBucket/settings.json
@@ -6,12 +6,12 @@
             "delegated": {
                 "read": [
                     {
-                        "name": "Tasks.Read"
+                        "name": "Tasks.Read.All"
                     }
                 ],
                 "update": [
                     {
-                        "name": "Tasks.ReadWrite"
+                        "name": "Tasks.ReadWrite.All"
                     }
                 ]
             },

--- a/Modules/Microsoft365DSC/DSCResources/MSFT_PlannerPlan/settings.json
+++ b/Modules/Microsoft365DSC/DSCResources/MSFT_PlannerPlan/settings.json
@@ -9,7 +9,7 @@
                         "name": "Group.Read.All"
                     },
                     {
-                        "name": "Tasks.Read"
+                        "name": "Tasks.Read.All"
                     }
                 ],
                 "update": [
@@ -17,10 +17,10 @@
                         "name": "Group.Read.All"
                     },
                     {
-                        "name": "Tasks.Read"
+                        "name": "Tasks.Read.All"
                     },
                     {
-                        "name": "Tasks.ReadWrite"
+                        "name": "Tasks.ReadWrite.All"
                     }
                 ]
             },

--- a/Modules/Microsoft365DSC/Modules/M365DSCPermissions.psm1
+++ b/Modules/Microsoft365DSC/Modules/M365DSCPermissions.psm1
@@ -1508,7 +1508,7 @@ function Update-M365DSCAzureAdApplication
                     $_ -is [Microsoft.Azure.PowerShell.Cmdlets.Resources.MSGraph.Models.ApiV10.MicrosoftGraphKeyCredential]
                 }
 
-                if ($PSBoundParameters.ContainsKey('CertificatePath'))
+                if (($PSBoundParameters.ContainsKey('CertificatePath') -and (-not $CreateSelfSignedCertificate)))
                 {
                     $cerCert = New-Object -TypeName System.Security.Cryptography.X509Certificates.X509Certificate2 -ArgumentList $CertificatePath
                 }

--- a/docs/docs/resources/planner/PlannerBucket.md
+++ b/docs/docs/resources/planner/PlannerBucket.md
@@ -32,11 +32,11 @@ To authenticate with the Microsoft Graph API, this resource required the followi
 
 - **Read**
 
-    - Tasks.Read
+    - Tasks.Read.All
 
 - **Update**
 
-    - Tasks.ReadWrite
+    - Tasks.ReadWrite.All
 
 #### Application permissions
 

--- a/docs/docs/resources/planner/PlannerPlan.md
+++ b/docs/docs/resources/planner/PlannerPlan.md
@@ -31,11 +31,11 @@ To authenticate with the Microsoft Graph API, this resource required the followi
 
 - **Read**
 
-    - Group.Read.All, Tasks.Read
+    - Group.Read.All, Tasks.Read.All
 
 - **Update**
 
-    - Group.Read.All, Tasks.Read, Tasks.ReadWrite
+    - Group.Read.All, Tasks.Read.All, Tasks.ReadWrite.All
 
 #### Application permissions
 

